JavaScript/JavaScriptSDK/Context/Session.ts

```diff
@@ -76,11 +79,14 @@ module Microsoft.ApplicationInsights.Context {
             // renew if acquisitionSpan or renewalSpan has ellapsed
             if (acquisitionExpired || renewalExpired) {
                 // update automaticSession so session state has correct id                
-                this.automaticSession.isFirst = undefined; 
+                this.automaticSession.isFirst = undefined;
                 this.renew();
             } else {
-                this.automaticSession.renewalDate = +new Date;
-                this.setCookie(this.automaticSession.id, this.automaticSession.acquisitionDate, this.automaticSession.renewalDate);
+                // do not update the cookie more often than cookieUpdateInterval
+                if (!this.cookieUpdatedTimestamp || now - this.cookieUpdatedTimestamp > _SessionManager.cookieUpdateInterval) {
+                    this.automaticSession.renewalDate = +new Date;
```

Comment:
I'm a member of the Microsoft Edge team and co-chair of the W3C Web Performance Working Group. Was forwarded this so pardon if we've not met! I recommend using performance.now rather than Date for things such as this. performance.now is monotonically increasing where Date can have skew.